### PR TITLE
Debug log for large overhead containers

### PR DESCRIPTION
**Pull request by adambalogh**

### CODE REVIEW

File: internal/extender/overhead.go [hunk 1]
```diff
@@ -146,10 +147,21 @@ func (o *OverheadComputer) compute(ctx context.Context) {
 	o.overheadLock.Unlock()
 }
 
-func podToResources(pod *v1.Pod) *resources.Resources {
+func podToResources(ctx context.Context, pod *v1.Pod) *resources.Resources {
+	oneCPU := resource.NewMilliQuantity(1000, resource.DecimalSI)
+	oneGiB := resource.NewQuantity(1*1024*1024*1024, resource.BinarySI)
```
Comment (onursatici) [inline internal/extender/overhead.go]: can these be constants?

File: internal/extender/overhead.go [hunk 2]
```diff
@@ -146,10 +147,21 @@ func (o *OverheadComputer) compute(ctx context.Context) {
...
+
 	res := resources.Zero()
 	for _, c := range pod.Spec.Containers {
-		res.AddFromResourceList(c.Resources.Requests)
+		resourceRequests := c.Resources.Requests
+		if resourceRequests.Cpu().AsDec().Cmp(oneCPU.AsDec()) > 0 || resourceRequests.Memory().AsDec().Cmp(oneGiB.AsDec()) > 0 {
+			svc1log.FromContext(ctx).Debug("Container with no resource reservation has high resource requests",
+				svc1log.SafeParam("Pod", pod.Name),
```
Comment (onursatici) [inline internal/extender/overhead.go]: can this be `podName` to match other logs?

Comment (adambalogh) [reply internal/extender/overhead.go]: 👍 fixed

File: internal/extender/overhead.go (hunk 1 above)
Comment (adambalogh) [reply internal/extender/overhead.go]: They can only be vars, not sure if that's better?

Comment (adambalogh) [reply internal/extender/overhead.go]: moved them outside

**Pull request merged**